### ISSUE TITLE
SES RawEmail - local address contains illegal character

### ISSUE THREAD

**Issue by aymanbagabas** [swift-aws/aws-sdk-swift#331]
**Describe the bug**
Unable to send raw emails. I keep getting `InvalidParameterValue: Local address contains illegal character`.

**To Reproduce**
1. Create a `RawMessage` from the example given in https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-raw.html#send-email-raw-mime `let message = SES.RawMessage(data: Data(rawMessage.utf8))`
2. Construct a RawEmailRequest `let request = SES.SendRawEmailRequest(destinations: ["To: recipient@example.com"], rawMessage: message, source: #""foo" <foo@bar.com>"#)`
3. Send the raw email `ses.sendRawEmail(request)`

```
let rawMessage = """
From: "foo" <foo@bar.com>
Subject: Customer service contact info
Content-Type: multipart/mixed;
    boundary="a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a"

--a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a
Content-Type: multipart/alternative;
    boundary="sub_a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a"

--sub_a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a
Content-Type: text/plain; charset=iso-8859-1
Content-Transfer-Encoding: quoted-printable

Please see the attached file for a list of customers to contact.

--sub_a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a
Content-Type: text/html; charset=iso-8859-1
Content-Transfer-Encoding: quoted-printable

<html>
<head></head>
<body>
<h1>Hello!</h1>
<p>Please see the attached file for a list of customers to contact.</p>
</body>
</html>

--sub_a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a--

--a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a
Content-Type: text/plain; name="customers.txt"
Content-Description: customers.txt
Content-Disposition: attachment;filename="customers.txt";
    creation-date="Sat, 05 Aug 2017 19:35:36 GMT";
Content-Transfer-Encoding: base64

SUQsRmlyc3ROYW1lLExhc3ROYW1lLENvdW50cnkKMzQ4LEpvaG4sU3RpbGVzLENhbmFkYQo5MjM4
OSxKaWUsTGl1LENoaW5hCjczNCxTaGlybGV5LFJvZHJpZ3VleixVbml0ZWQgU3RhdGVzCjI4OTMs
QW5heWEsSXllbmdhcixJbmRpYQ==

--a3f166a86b56ff6c37755292d690675717ea3cd9de81228ec2b76ed4a15d6d1a--
"""
let message = SES.RawMessage(data: Data(rawMessage.utf8))
let request = SES.SendRawEmailRequest(
            destinations: ["To: recipient@example.com"],
            rawMessage: message,
            source: "\"foo\" <foo@bar.com>"
)
ses.sendRawEmail(request).map { print($0) }.catchMap { print($0) }
```

**Expected behavior**
An email is sent with content and attachments.

**Setup (please complete the following information):**
 - OS: macOS 10.15.5
 - Version of aws-sdk-swift: 3.5.1
 - Authentication mechanism: hard-coded credentials

**Comment by adam-fowler** [comment]
Is it because you have speech marks around foo in the source address?

**Comment by adam-fowler** [comment]
Remove the "To:" from your destination string

**Comment by aymanbagabas** [comment]
> Remove the "To:" from your destination string

Now I get an empty email, no subject, no content, and it displays the email instead of the name specified in `source`. The raw source email shows `To: Undisclosed recipients:;`.

**Comment by adam-fowler** [comment]
I had it working fine for me. When I said remove the “To:”. I hope you didn’t edit the raw email, just the destinations parameter of the SendRawEmailRequest object.

**Comment by aymanbagabas** [comment]
> I had it working fine for me. When I said remove the “To:”. I hope you didn’t edit the raw email, just the destinations parameter of the SendRawEmailRequest object.

That's exactly what I did, I removed the "To:" from the destinations array. According to the api [reference](https://swift-aws.github.io/aws-sdk-swift-docs/4.x.x/Structs/SES/SendRawEmailRequest.html#/s:3SESAAV19SendRawEmailRequestV12destinationsSaySSGSgvp) `destinations` is "A list of destinations for the message, consisting of To:, CC:, and BCC: addresses." If I understand this correctly, these should be prepended to the addresses, or not?

**Comment by adam-fowler** [comment]
Not totally sure, but as soon as I removed the "To: " it worked. Looking at your code above you need a "To: " entry in the header of your raw email though, that'll be why you are getting a `To: Undisclosed recipients:`.

**Comment by adam-fowler** [comment]
Yes verified your raw email needs to To:, Cc: or Bcc: header, otherwise it is not considered a valid email. The `destinations` parameter can be used to override who the email is actually sent to.

**Comment by aymanbagabas** [comment]
Looking at the email source, I notice this header `Authentication-Results: spf=pass (sender IP is <ipaddress>)
 smtp.mailfrom=amazonses.com; <domain>; dkim=fail (signature did not
 verify) header.d=<domain>;<domain>; dmarc=none action=none
 header.from=<domain>;compauth=fail reason=601`. But when using `sendEmail` it works fine and I get `compauth=pass reason=109`.

Also noticed that when omitting the `source` and/or `destination` and using `From`/`To` headers in the raw message, it complains about missing header `InvalidParameterValue: Missing required header 'From'/'To'.`

**Comment by adam-fowler** [comment]
Ah worked out what the issue is. SendRawEmail in v3.x.x of AWSSDKSwift doesnt work. It has been fixed in 4.x.x (https://github.com/swift-aws/aws-sdk-swift-core/pull/216). Are you required to work on 3.x.x or can you upgrade.

**Comment by adam-fowler** [comment]
If you update and get aws-sdk-swift-core v3.5.2 this should be fixed. With this fix you should be able to specify the `To` and `From` addresses in either the email header or the `SendRawEmailRequest` struct (do not include the "To: " prefix). Although it is best if you do include them in the raw email. 

Can you verify this all works for you

**Comment by aymanbagabas** [comment]
@adam-fowler I can confirm that this is fixed with aws-sdk-swift-core v3.5.2. Thanks for your hard work!
